src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultRazorDocumentMappingService.cs

```diff
@@ -77,6 +78,14 @@ public override bool TryMapToProjectedDocumentRange(RazorCodeDocument codeDocume
                 return false;
             }
 
+            // Ensures a valid range is returned
```

Comment:
SHould expand a little more to indicate why we can ever generate an "invalid" range. Also if it's easy to add a test for this I would

Reply:
Added description and more testing